src/Data/Fin/Relation/Unary/Top.agda

```diff
@@ -0,0 +1,135 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The 'top' view of Fin
+--
+-- This is an example of a view of (elements of) a datatype,
+-- here i : Fin (suc n), which exhibits every such i as
+-- * either, i = fromℕ n
+-- * or, i = inject₁ j for a unique j : Fin n
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Fin.Relation.Unary.Top where
+
+open import Data.Nat.Base using (ℕ; zero; suc; _<_)
+open import Data.Fin.Base using (Fin; zero; suc; toℕ; fromℕ; inject₁)
+open import Data.Fin.Properties using (toℕ-fromℕ; toℕ-inject₁; inject₁ℕ<)
+open import Relation.Binary.PropositionalEquality
+
+private
+  variable
+    n : ℕ
+
+------------------------------------------------------------------------
+-- The View, considered as a unary relation on Fin (suc n)
+
+-- NB `Data.Fin.Properties.fromℕ≢inject₁` establishes that the following
+-- inductively defined family on `Fin (suc n)` has constructors which
+-- target *disjoint* instances of the family; and hence the interpretations
+-- of the View constructors will also be disjoint
+
+data View : (i : Fin (suc n)) → Set where
+
+  top :                View (fromℕ n)
+  inj : (j : Fin n) → View (inject₁ j)
+
+-- The view covering function, witnessing soundness of the view
+
+view : ∀ {n} i → View {n} i
```

Comment:
I'll look again at this, but I have found Agda to be pretty picky about `variable` scope (and I'm trying to avoid giving an explicit type to `i`). I think, but will re-check, that it complained when I wrote
```agda
view : ∀ i → View {n} i
```
and then wrote the definition based on *pattern matching* on `n`.

My belief is that `{n}` is correct, because in any *use* of `view`, `n` will already be known/inferrable. But I will go through it again and see.

Reply:
Also, my instinct is always to provide `{n}` as a nudge to the typechecker, in place of the verbosity of quantifying over `(i : Fin (suc n))`, cf. other PRs where we have differed as to explicit quantification vs. supplying enough implicits to satisfy the typechecker.

Reply:
Final revision to the type of the `view` function, as per your suggestion